core/src/main/kotlin/maru/consensus/ForkId.kt

```diff
@@ -8,64 +8,48 @@
  */
 package maru.consensus
 
-import maru.core.Hasher
-import maru.database.BeaconChain
-import maru.serialization.Serializer
+import java.nio.ByteBuffer
+import java.util.zip.CRC32
+import org.apache.tuweni.bytes.Bytes
 
 data class ForkId(
   val chainId: UInt,
-  val forkSpec: ForkSpec,
   val genesisRootHash: ByteArray,
 ) {
+  companion object {
+    const val FORK_ID_FIELD_NAME = "mfid"
+  }
+
+  val bytes = computeBytes()
+
   override fun equals(other: Any?): Boolean {
     if (this === other) return true
     if (javaClass != other?.javaClass) return false
 
     other as ForkId
 
     if (chainId != other.chainId) return false
-    if (forkSpec != other.forkSpec) return false
     if (!genesisRootHash.contentEquals(other.genesisRootHash)) return false
 
     return true
   }
 
   override fun hashCode(): Int {
     var result = chainId.hashCode()
-    result = 31 * result + forkSpec.hashCode()
     result = 31 * result + genesisRootHash.contentHashCode()
     return result
   }
-}
 
-class ForkIdHasher(
-  val forkIdSerializer: Serializer<ForkId>,
-  val hasher: Hasher,
-) {
-  fun hash(forkId: ForkId): ByteArray = hasher.hash(forkIdSerializer.serialize(forkId)).takeLast(4).toByteArray()
-}
-
-class ForkIdHashProvider(
-  private val chainId: UInt,
-  private val beaconChain: BeaconChain,
-  private val forksSchedule: ForksSchedule,
-  private val forkIdHasher: ForkIdHasher,
-) {
-  fun currentForkIdHash(): ByteArray {
-    val forkId =
-      ForkId(
-        chainId = chainId,
-        forkSpec =
-          forksSchedule.getForkByTimestamp(
```

Comment:
While having many abstractions vs putting all the Fork Id computation logic into a single function can be debateable, the substantial difference I see here, is that with your changes fork spec is not participating in the checksum computation. This means that nodes with with different protocol rules (block time, fork switch timestamp, initial validator set) may peer to each other. I wanted to avoid that